mllib/src/main/scala/org/apache/spark/ml/regression/AFTSurvivalRegression.scala

```diff
@@ -0,0 +1,400 @@
+    copyValues(model.setParent(this))
+  }
+
+  override def transformSchema(schema: StructType): StructType = {
+    validateAndTransformSchema(schema, fitting = true)
+  }
+
+  override def copy(extra: ParamMap): AFTSurvivalRegression = defaultCopy(extra)
+}
+
+/**
+ * :: Experimental ::
+ * Model produced by [[AFTSurvivalRegression]].
+ */
+@Experimental
+class AFTSurvivalRegressionModel private[ml] (
+    override val uid: String,
+    val weights: Vector,
+    val intercept: Double,
+    val scale: Double)
+  extends Model[AFTSurvivalRegressionModel] with AFTSurvivalRegressionParams {
+
+  /** @group setParam */
+  def setFeaturesCol(value: String): this.type = set(featuresCol, value)
+
+  /** @group setParam */
+  def setPredictionCol(value: String): this.type = set(predictionCol, value)
+
+  /** @group setParam */
+  def setQuantile(value: Vector): this.type = set(quantile, value)
+
+  def quantilePredict(features: Vector): Vector = {
+    require(hasQuantile, "AFTSurvivalRegressionModel quantilePredict must set quantile vector")
+    // scale parameter of the Weibull distribution
+    val lambda = math.exp(weights.toBreeze.dot(features.toBreeze) + intercept)
+    // shape parameter of the Weibull distribution
+    val k = 1 / scale
+    val array = $(quantile).toArray.map { q => lambda * math.exp(math.log(-math.log(1-q)) / k) }
+    Vectors.dense(array)
+  }
+
+  def predict(features: Vector): Double = {
+    val lambda = math.exp(weights.toBreeze.dot(features.toBreeze) + intercept)
+    lambda
+  }
+
+  override def transform(dataset: DataFrame): DataFrame = {
+    transformSchema(dataset.schema)
+    val predictUDF = udf { features: Vector => predict(features) }
+    dataset.withColumn($(predictionCol), predictUDF(col($(featuresCol))))
+  }
+
+  override def transformSchema(schema: StructType): StructType = {
+    validateAndTransformSchema(schema, fitting = false)
+  }
+
+  override def copy(extra: ParamMap): AFTSurvivalRegressionModel = {
+    copyValues(new AFTSurvivalRegressionModel(uid, weights, intercept, scale), extra)
+      .setParent(parent)
+  }
+}
+
+/**
+ * AFTAggregator computes the gradient and loss for a AFT loss function,
+ * as used in AFT survival regression for samples in sparse or dense vector in a online fashion.
+ *
+ * The loss function and likelihood function under the AFT model based on:
+ * Lawless, J. F., Statistical Models and Methods for Lifetime Data,
+ * New York: John Wiley & Sons, Inc. 2003.
+ *
+ * Two AFTAggregator can be merged together to have a summary of loss and gradient of
+ * the corresponding joint dataset.
+ *
+ * For random lifetime T_{i} of subjects i = 1, ..., n, with possible right-censoring,
+ * the likelihood function under the AFT model is given as
+ * {{{
+ *   L(\beta,\sigma)=\prod_{i=1}^n[\frac{1}{\sigma}f_{0}
+ *   (\frac{\log{t_{i}}-x^{'}\beta}{\sigma})]^{\delta_{i}}S_{0}
+ *   (\frac{\log{t_{i}}-x^{'}\beta}{\sigma})^{1-\delta_{i}}
+ * }}}
```

Comment:
1) delta in the formula is the *event* indicator. Since we decide to use the *censored* indicator, I recommend we clarify this in the formula.

2) please add a sentence to clarify t_{i} = log(T_{i}).